test/e2e/unmanaged_dns_test.go

```diff
@@ -186,6 +192,23 @@ func TestManagedDNSToUnmanagedDNSIngressController(t *testing.T) {
 func TestUnmanagedDNSToManagedDNSInternalIngressController(t *testing.T) {
 	t.Parallel()
 
+	if infraConfig.Status.PlatformStatus == nil {
+		t.Skip("test skipped on nil platform")
+	}
+	platform := infraConfig.Status.PlatformStatus.Type
+
+	supportedPlatforms := map[configv1.PlatformType]struct{}{
+		configv1.AlibabaCloudPlatformType: {},
+		configv1.AWSPlatformType:          {},
+		configv1.AzurePlatformType:        {},
+		configv1.GCPPlatformType:          {},
+		configv1.IBMCloudPlatformType:     {},
+		configv1.PowerVSPlatformType:      {},
+	}
+	if _, supported := supportedPlatforms[platform]; !supported {
+		t.Skipf("test skipped on platform %q", platform)
+	}
```

Comment:
I feel like over time, this will cause problems.  When a new platform is added, we will have to remember to add it here.  If there is a specific platform that we don't want to run this on, you could do this:

```suggestion
	unsupportedPlatforms := map[configv1.PlatformType]struct{}{
		configv1.someSpecificPlatformType: {},
	}
	if _, unsupported := unsupportedPlatforms[platform]; unsupported {
		t.Skipf("test skipped on platform %q", platform)
	}
```

Reply:
Yea I see your point, opt-out vs. opt-in. @Miciah wrote this, and I think he wanted to be explicit with an "opt-in" approach to this E2E test.

@Miciah what are your thoughts since you added this? What platforms are we really trying to skip here (Nutanix? Libvirt? Openstack?...) can we even run e2e-test on them?

Reply:
We're trying to skip platforms that don't use or support LoadBalancer-type services.  We have the same or similar logic repeated in several tests.  This should probably be refactored as part of a general cleanup of how we handle platform-specific skips.

Reply:
@Miciah Yea I think your point is that this is a larger issue than just this single E2E test. Should we just remove this logic for the time being in anticipation for a more comprehensive effort? (kind of the same debate with Andy's comment on retries https://github.com/openshift/cluster-ingress-operator/pull/845/files#r999688986)

Reply:
I don't see a reason to remove this logic—it is strictly better to skip this test on platforms on which we know the test would fail than to run the test and fail on those platforms.

Reply:
Before I commented, I searched for uses of this pattern, and I didn't find any in c-i-o, except in the case where we knew we only wanted to run the test on a single platform.  https://github.com/openshift/cluster-ingress-operator/search?q=supportedplatforms

Reply:
GitHub's search is unreliable.  The pattern appears in several tests.

- `TestAllowedSourceRanges`: https://github.com/openshift/cluster-ingress-operator/blob/9456d9fade4bf9afab26a26e089b6c44a51bc444/test/e2e/allowed_source_ranges_test.go#L28-L41
- `TestAllowedSourceRangesStatus`: https://github.com/openshift/cluster-ingress-operator/blob/9456d9fade4bf9afab26a26e089b6c44a51bc444/test/e2e/allowed_source_ranges_test.go#L127-L140
- `TestSourceRangesProgressingAndEvaluationConditionsDetectedStatuses`: https://github.com/openshift/cluster-ingress-operator/blob/9456d9fade4bf9afab26a26e089b6c44a51bc444/test/e2e/allowed_source_ranges_test.go#L246-L259
- `TestInternalLoadBalancer`: https://github.com/openshift/cluster-ingress-operator/blob/9456d9fade4bf9afab26a26e089b6c44a51bc444/test/e2e/operator_test.go#L935-L954
- `TestInternalLoadBalancerGlobalAccessGCP`: https://github.com/openshift/cluster-ingress-operator/blob/9456d9fade4bf9afab26a26e089b6c44a51bc444/test/e2e/operator_test.go#L1029-L1039
- `TestScopeChange`: https://github.com/openshift/cluster-ingress-operator/blob/9456d9fade4bf9afab26a26e089b6c44a51bc444/test/e2e/operator_test.go#L1299-L1315
- `TestLocalWithFallbackOverrideForLoadBalancerService`: https://github.com/openshift/cluster-ingress-operator/blob/9456d9fade4bf9afab26a26e089b6c44a51bc444/test/e2e/operator_test.go#L2879-L2890